lib/sudo-exec/src/monitor.rs

```diff
@@ -74,38 +78,35 @@ impl MonitorRelay {
     pub(super) fn run(mut self) -> io::Result<std::convert::Infallible> {
         loop {
             // First we check if the command is finished
-            self.wait_command()?;
-
-            // Then we check any pending signals that we received. Based on `mon_signal_cb`
-            for info in self.signals.wait() {
-                self.relay_signal(info);
-            }
-        }
-    }
+            if let Some(status) = self.wait_command()? {
+                ExitReason::from_status(status).send(&self.tx)?;
 
-    fn wait_command(&mut self) -> io::Result<()> {
-        if let Some(status) = self.command.try_wait()? {
-            ExitReason::from_status(status).send(&self.tx)?;
+                // Given that we overwrote the default handlers for all the signals, we must
+                // emulate them to handle the signal we just sent correctly.
+                drop(self.signal_handlers);
```

Comment:
this changes the `SignalAction` from `Stream` to `Default` but does not necessarily ensure that any pending signals (buffered in the `UnixStream::Pair`) are handled using the default_signal_handler like the old version did, right?

I don't think how the loop is written guarantees that there are no pending signals when this line is reached. there could have been pending signals on the first iteration of this loop, before `signal_handlers.poll` is called

Reply:
That's a good point!

Maybe I could add a `pending` method that takes the handlers by value and:
- Sets all the sockets to non-blocking mode.
- Tries to read from them and store all the `Ok(SignalInfo)` values that we can get.
- Return the `SignalInfo` values in a `Vec`.
In that way the handlers will be `drop`ped so we set the action to `Default`  and we can emulate the default handler for the returned `SignalInfo` values.

Reply:
that should work

Reply:
Hmmm... the comment here makes me think that this is not even necessary... The only purpose of sending signals to the monitor is for them to be forwarded to the command which has already exited in one way or another.

This might be just some remnant of the `!use_pty` event loop we used to have. I'm going to remove this `drop` and the comment all-together.